File: Android /Android/25:Jan:2017/FishbowlMobileApps/AndroidSDK/FishbowlMobileApps-Jamba-Android-v2.4.1/JMBA_Kiosk_iOS/JambaKiosk/AuditOrder.swift
//
//  AuditOrderStatus.swift
//  JambaJuice
//
//  Created by Eneko Alonso on 8/31/15.
//  Copyright (c) 2015 Jamba Juice. All rights reserved.
//

import Foundation
import Parse

struct AuditOrder {

    static let parseClassName = "AuditOrder"

    private var basket: Basket
    private var orderStatus: OrderStatus
    private var user: User?

    init(basket: Basket, orderStatus: OrderStatus, user: User?) {
        self.basket = basket
        self.orderStatus = orderStatus
        self.user = user
    }

    func serializeAsParseObject() -> PFObject {
        let parseObject = PFObject(className: AuditOrder.parseClassName)

        // Order Details
        parseObject["orderId"]         = orderStatus.orderStatusId
        parseObject["vendorName"]      = orderStatus.vendorName
        parseObject["vendorExtRef"]    = orderStatus.vendorExtRef
        parseObject["orderReadyTime"]  = orderStatus.readyTime ?? NSNull()
        parseObject["orderPlacedTime"] = orderStatus.timePlaced ?? NSNull()
        parseObject["orderTotal"]      = orderStatus.total
        parseObject["orderSubtotal"]   = orderStatus.subTotal
        parseObject["orderTax"]        = orderStatus.salesTax
        parseObject["productCount"]    = orderStatus.products.count

        // User details
        parseObject["userId"]           = user?.userId ?? NSNull()
        parseObject["userEmailAddress"] = user?.emailAddress ?? NSNull()
        parseObject["userPhone"]        = user?.phoneNumber ?? NSNull()
        parseObject["userFirstName"]    = user?.firstName ?? NSNull()
        parseObject["userLastName"]     = user?.lastName ?? NSNull()
        parseObject["userDateOfBirth"]  = NSNull()

        return parseObject
    }

}

File: Android /Android/25:Jan:2017/FishbowlMobileApps/AndroidSDK/FishbowlMobileApps-Jamba-Android-v2.4.2/JMBA_JambaJuice_Android/app/src/main/java/com/olo/jambajuice/BusinessLogic/Models/AuditOrder.java
package com.olo.jambajuice.BusinessLogic.Models;

import com.olo.jambajuice.BusinessLogic.Services.UserService;
import com.olo.jambajuice.Utils.ParseUtils;
import com.parse.ParseObject;

import java.text.SimpleDateFormat;
import java.util.Date;

import static com.wearehathway.apps.olo.Utils.Constants.Server_Time_Format;

/**
 * Created by Nauman Afzaal on 01/09/15.
 */
public class AuditOrder
{
    public static String parseClassName = "AuditOrder";

    Basket basket;
    OrderStatus orderStatus;
    User user;

    public AuditOrder(Basket basket, OrderStatus orderStatus, User user)
    {
        this.basket = basket;
        this.orderStatus = orderStatus;
        this.user = user;
    }

    public ParseObject serializeAsParseObject()
    {
        ParseObject parseObject = ParseObject.create(AuditOrder.parseClassName);

        // Order Details
        parseObject.put("orderId", ParseUtils.sanitizeValue(orderStatus.getId()));
        parseObject.put("vendorName", ParseUtils.sanitizeValue(orderStatus.getVendorname()));
        parseObject.put("vendorExtRef", ParseUtils.sanitizeValue(orderStatus.getVendorExtRef()));
        SimpleDateFormat format = new SimpleDateFormat(Server_Time_Format);
        Date placedTime = new Date();
        Date readyTime = new Date();
        try
        {
            placedTime = format.parse(orderStatus.getTimePlaced());
            readyTime = format.parse(orderStatus.getReadytime());
        }
        catch (Exception e)
        {
            e.printStackTrace();
        }
        parseObject.put("orderPlacedTime", ParseUtils.sanitizeValue(placedTime));
        parseObject.put("orderReadyTime", ParseUtils.sanitizeValue(readyTime));
        parseObject.put("orderTotal", ParseUtils.sanitizeValue(orderStatus.getTotal()));
        parseObject.put("orderSubtotal", ParseUtils.sanitizeValue(orderStatus.getSubTotal()));
        parseObject.put("orderTax", ParseUtils.sanitizeValue(orderStatus.getSaleTax()));
        parseObject.put("productCount", ParseUtils.sanitizeValue(orderStatus.getProductsCount()));

        // User details
        if (UserService.isUserAuthenticated())
        {
            parseObject.put("userId", ParseUtils.sanitizeValue(user.getSpendGoId()));
            parseObject.put("userEmailAddress", ParseUtils.sanitizeValue(user.getEmailaddress()));
            parseObject.put("userPhone", ParseUtils.sanitizeValue(user.getContactnumber()));
            parseObject.put("userFirstName", ParseUtils.sanitizeValue(user.getFirstname()));
            parseObject.put("userLastName", ParseUtils.sanitizeValue(user.getLastname()));
            parseObject.put("userDateOfBirth", ParseUtils.sanitizeValue(user.getDob()));
        }

        return parseObject;
    }
}
